src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -280,19 +295,10 @@ private void DoClose()
 		{
 			if (m_connectionState != ConnectionState.Closed)
 			{
-				m_cachedProcedures = null;
-				if (ActiveReader != null){
-					ActiveReader.Dispose();
-					ActiveReader = null;
-				}
-				if (CurrentTransaction != null)
-				{
-					CurrentTransaction.Dispose();
-					CurrentTransaction = null;
-				}
+				CloseDatabase();
 				if (m_session != null)
 				{
-					if (m_connectionSettings.Pooling)
+					if (m_connectionSettings.Pooling && m_changedDatabase == null)
```

Comment:
I believe this will cause this connection to be "Checked Out" of the pool forever, diminishing the available size of the pool.  I think that a better implementation would be to mark the connection "dirty" and throw it away in [ConnectionPool.cs#L61](https://github.com/mysql-net/MySqlConnector/blob/master/src/MySqlConnector/MySqlClient/ConnectionPool.cs#L61), much like we do when we check `session.PoolGeneration`